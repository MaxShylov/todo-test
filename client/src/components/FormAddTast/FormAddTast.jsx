import React, { Component } from 'react';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Input from 'antd/lib/input';
import Button from 'antd/lib/button/button';
import Container from '../Container/Container';

import './FormAddTast.scss'
import { addTask, saveTasks } from '../../redux/actions/tasks/tasks';

const { func } = PropTypes;
const InputGroup = Input.Group;


export class FormAddTast extends Component {
  static propTypes = {
    addTask: func,
    saveTasks: func
  };

  state = { value: null };

  onChange = (e) => this.setState({ value: e.target.value });

  addTask = () => {
    const
      { addTask, saveTasks } = this.props,
      { value } = this.state;

    if (!value) return;

    let id = Date.now();

    if (process.env.NODE_ENV === 'test') id = 1;

    addTask({
      id,
      isDone: false,
      content: this.state.value
    });

    saveTasks();
    this.setState({ value: null })
  };

  render() {
    const { value } = this.state;

    return (
      <Container className="FormAddTast">
        <InputGroup compact>
          <Input
            value={value}
            onChange={this.onChange}
            onPressEnter={this.addTask}
          />
          <Button onClick={this.addTask}>
            Add task
          </Button>
        </InputGroup>
      </Container>
    );
  }
}


const matchDispatchToProps = dispatch => bindActionCreators({
  addTask,
  saveTasks
}, dispatch);

export default connect(null, matchDispatchToProps)(FormAddTast);
