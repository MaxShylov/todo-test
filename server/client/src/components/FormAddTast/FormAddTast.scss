@import "../../styles/variables";

.FormAddTast {
  display: flex;
  padding-top: 50px;

  .ant-input-group {
    display: flex;
  }
}
