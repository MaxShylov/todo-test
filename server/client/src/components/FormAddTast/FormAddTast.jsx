import React, { Component } from 'react';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Input from 'antd/lib/input';
import Button from 'antd/lib/button/button';
import Container from '../Container/Container';

import './FormAddTast.scss'
import { addTask } from '../../redux/actions/tasks/tasks';

const { func } = PropTypes;
const InputGroup = Input.Group;


export class FormAddTast extends Component {
  static propTypes = {
    addTask: func
  };

  state = { value: null };

  onChange = (e) => this.setState({ value: e.target.value });

  addTask = () => {
    const { value } = this.state;

    if (!value) return;

    this.props.addTask({
      id: Date.now(),
      isDone: false,
      content: this.state.value
    });

    this.setState({ value: null })
  };

  render() {
    const { value } = this.state;

    return (
      <Container className="FormAddTast">
        <InputGroup compact>
          <Input
            value={value}
            onChange={this.onChange}
            onPressEnter={this.addTask}
          />
          <Button onClick={this.addTask}>
            Add task
          </Button>
        </InputGroup>
      </Container>
    );
  }
}


const matchDispatchToProps = dispatch => bindActionCreators({
  addTask
}, dispatch);

export default connect(null, matchDispatchToProps)(FormAddTast);
