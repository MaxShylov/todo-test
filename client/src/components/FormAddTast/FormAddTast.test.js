import React from 'react';
import { shallow } from 'enzyme'

import { FormAddTast } from './FormAddTast.jsx'


describe('FormAddTast', () => {

  describe('initial', () => {
    const wrapper = shallow(<FormAddTast />);

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
  });

  describe('method onChange', () => {
    const
      wrapper = shallow(<FormAddTast />),
      value = 123;

    wrapper.find('Input').simulate('change', { target: { value } });

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('state value changed', () => expect(wrapper.state().value).toEqual(value));
  });

  describe('method addTask without state.value', () => {
    const
      mockAddTask = jest.fn(),
      mockSaveTasks = jest.fn(),
      props = {
        addTask: mockAddTask,
        saveTasks: mockSaveTasks
      },
      wrapper = shallow(<FormAddTast {...props} />);

    wrapper.find('Button').simulate('click');

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('addTask didn\'t call', () => expect(mockAddTask).not.toHaveBeenCalled());
    it('saveTasks didn\'t call', () => expect(mockSaveTasks).not.toHaveBeenCalled());
  });

  describe('method addTask with state.value', () => {
    const
      mockAddTask = jest.fn(),
      mockSaveTasks = jest.fn(),
      props = {
        addTask: mockAddTask,
        saveTasks: mockSaveTasks
      },
      wrapper = shallow(<FormAddTast {...props} />);

    wrapper.setState({ value: 123 });

    wrapper.find('Button').simulate('click');

    it('renders properly', () => expect(wrapper).toMatchSnapshot());
    it('addTask called', () => expect(mockAddTask).toHaveBeenCalled());
    it('saveTasks called', () => expect(mockSaveTasks).toHaveBeenCalled());
    it('state value changed', () => expect(wrapper.state().value).isNull);
  });

});
