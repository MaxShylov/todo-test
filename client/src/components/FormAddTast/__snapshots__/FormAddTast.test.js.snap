// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAddTast initial renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormAddTast />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Group
        compact={true}
      >
        <Input
          disabled={false}
          onChange={[Function]}
          onPressEnter={[Function]}
          type="text"
          value={null}
        />
        <Button
          block={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
        >
          Add task
        </Button>
      </Group>,
      "className": "FormAddTast",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={null}
          />,
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>,
        ],
        "compact": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "onChange": [Function],
            "onPressEnter": [Function],
            "type": "text",
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "Add task",
            "ghost": false,
            "loading": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add task",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Group
          compact={true}
        >
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={null}
          />
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>
        </Group>,
        "className": "FormAddTast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Input
              disabled={false}
              onChange={[Function]}
              onPressEnter={[Function]}
              type="text"
              value={null}
            />,
            <Button
              block={false}
              ghost={false}
              loading={false}
              onClick={[Function]}
            >
              Add task
            </Button>,
          ],
          "compact": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "onChange": [Function],
              "onPressEnter": [Function],
              "type": "text",
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": "Add task",
              "ghost": false,
              "loading": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add task",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`FormAddTast method addTask with state.value renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormAddTast
    addTask={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "content": 123,
              "id": 1,
              "isDone": false,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    saveTasks={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Group
        compact={true}
      >
        <Input
          disabled={false}
          onChange={[Function]}
          onPressEnter={[Function]}
          type="text"
          value={null}
        />
        <Button
          block={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
        >
          Add task
        </Button>
      </Group>,
      "className": "FormAddTast",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={null}
          />,
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>,
        ],
        "compact": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "onChange": [Function],
            "onPressEnter": [Function],
            "type": "text",
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "Add task",
            "ghost": false,
            "loading": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add task",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Group
          compact={true}
        >
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={null}
          />
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>
        </Group>,
        "className": "FormAddTast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Input
              disabled={false}
              onChange={[Function]}
              onPressEnter={[Function]}
              type="text"
              value={null}
            />,
            <Button
              block={false}
              ghost={false}
              loading={false}
              onClick={[Function]}
            >
              Add task
            </Button>,
          ],
          "compact": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "onChange": [Function],
              "onPressEnter": [Function],
              "type": "text",
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": "Add task",
              "ghost": false,
              "loading": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add task",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`FormAddTast method addTask without state.value renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormAddTast
    addTask={[MockFunction]}
    saveTasks={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Group
        compact={true}
      >
        <Input
          disabled={false}
          onChange={[Function]}
          onPressEnter={[Function]}
          type="text"
          value={null}
        />
        <Button
          block={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
        >
          Add task
        </Button>
      </Group>,
      "className": "FormAddTast",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={null}
          />,
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>,
        ],
        "compact": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "onChange": [Function],
            "onPressEnter": [Function],
            "type": "text",
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "Add task",
            "ghost": false,
            "loading": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add task",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Group
          compact={true}
        >
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={null}
          />
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>
        </Group>,
        "className": "FormAddTast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Input
              disabled={false}
              onChange={[Function]}
              onPressEnter={[Function]}
              type="text"
              value={null}
            />,
            <Button
              block={false}
              ghost={false}
              loading={false}
              onClick={[Function]}
            >
              Add task
            </Button>,
          ],
          "compact": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "onChange": [Function],
              "onPressEnter": [Function],
              "type": "text",
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": "Add task",
              "ghost": false,
              "loading": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add task",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`FormAddTast method onChange renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormAddTast />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Group
        compact={true}
      >
        <Input
          disabled={false}
          onChange={[Function]}
          onPressEnter={[Function]}
          type="text"
          value={123}
        />
        <Button
          block={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
        >
          Add task
        </Button>
      </Group>,
      "className": "FormAddTast",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={123}
          />,
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>,
        ],
        "compact": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "onChange": [Function],
            "onPressEnter": [Function],
            "type": "text",
            "value": 123,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "Add task",
            "ghost": false,
            "loading": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add task",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Group
          compact={true}
        >
          <Input
            disabled={false}
            onChange={[Function]}
            onPressEnter={[Function]}
            type="text"
            value={123}
          />
          <Button
            block={false}
            ghost={false}
            loading={false}
            onClick={[Function]}
          >
            Add task
          </Button>
        </Group>,
        "className": "FormAddTast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Input
              disabled={false}
              onChange={[Function]}
              onPressEnter={[Function]}
              type="text"
              value={123}
            />,
            <Button
              block={false}
              ghost={false}
              loading={false}
              onClick={[Function]}
            >
              Add task
            </Button>,
          ],
          "compact": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "onChange": [Function],
              "onPressEnter": [Function],
              "type": "text",
              "value": 123,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": "Add task",
              "ghost": false,
              "loading": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add task",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
